### ISSUE TITLE
Quick check-in w r t supported functionality

### ISSUE THREAD

**Issue by jgarplind** [unomed-dev/react-native-matrix-sdk#20]
I was made aware of your project through https://matrix.org/blog/2024/12/06/this-week-in-matrix-2024-12-06/ where you mentioned:

> We have successfully used this library to prototype a cross-platform app including room list and timeline but it is a fairly young project and will most definitely still have a few sharp edges.

On my end, I have a fairly simple client based on matrix-js-sdk, which I was hoping to make available for React Native apps too, but as you may be aware, that SDK is not compatible with React Native (or so most of the discussion in that space would have me believe), among other things because it relies on IndexedDB.

So I just wanted to check in before spending serious time on giving this library a shot, are there any major functions missing here as compared to the JS SDK? Outside of the very basics, I'm interested in things such as:
- Auto-joining rooms I've been invited to
- Read receipts
- Typing state
- `scrollback` or other mechanisms to fetch historical messages
- Notifications, setting up rules and surfacing unread message notifications
- Looking up alias from roomId and vice versa

If there is any convenient way (short of reading through all the generated files) to find out what functionality is supported, I would gladly find out by myself.

Finally, thanks for working on this! I believe that there is a big need for a solid and maintained React Native client library.

**Comment by Johennes** [comment]
I've also spent a lot of time making the JS SDK work in React Native. There's nothing wrong with the JS SDK itself. It's more that React Native's environment is so peculiar that the team behind the JS SDK would have to spend a fair amount of time to keep it compatible.

To your question, this SDK wraps the FFI bindings of matrix-rust-sdk. The latter are being developed for and primarily used in the Element X apps. So a quick way to find out whether some functionality is supported is to check if Element X has it.

From the top of my head everything on your list should be included.

However, being a younger project, the Rust SDK will still have fewer features than the JS SDK. For instance, I believe the threads support in the Rust SDK is more limited. The other thing to be aware of is that the FFI bindings expose a subset of the Rust SDK only. For instance, when I started on this project, I needed legacy SSO support. The Rust SDK has it but the bindings did not because Element X doesn't use legacy SSO. So I had to submit a pull request to expose this feature in the bindings.
